src/utils/getSearchParamsFromRequest.js

```diff
@@ -0,0 +1,27 @@
+export function getSearchParamsFromRequest(request, defaultValues = {}, transform = null) {
+  if (!request) return undefined;
+  if (!request.url) return undefined;
+  try {
+    const url = new URL(request.url);
+    const searchParams = {};
+    url.searchParams.forEach((val, key) => {
```

Comment:
🟠 isn't there is a lib for such functionality?

Reply:
I didn't find a function that would parse query params from request in Next.js documentation. As for a new package, i don't think it's worth to search for a package for a simple function like this. Besides, i added additional features such as default values and transform, which i would need to implement too if i found a custom package